sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/OpenTelemetrySdkAutoConfiguration.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.autoconfigure;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.metrics.SdkMeterProvider;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+/**
+ * Auto-configuration for the OpenTelemetry SDK. As an alternative to programmatically configuring
+ * the SDK using {@link OpenTelemetrySdk#builder()}, this package can be used to automatically
+ * configure the SDK using environment properties specified by OpenTelemetry.
+ */
+public final class OpenTelemetrySdkAutoConfiguration {
+
+  /**
+   * Returns an {@link OpenTelemetrySdk} automatically initialized through recognized system
+   * properties and environment variables.
+   */
+  public static OpenTelemetrySdk initialize() {
+    ConfigProperties config = ConfigProperties.get();
+    ContextPropagators propagators = PropagatorConfiguration.configurePropagators(config);
+
+    Resource resource = configureResource(config);
+
+    Set<String> exporterNames =
+        new LinkedHashSet<>(config.getCommaSeparatedValues("otel.exporter"));
+
+    configureMeterProvider(resource, exporterNames, config);
+
+    SdkTracerProvider tracerProvider =
+        TracerProviderConfiguration.configureTracerProvider(resource, exporterNames, config);
+
+    if (!exporterNames.isEmpty()) {
```

Comment:
I really really don't like that the previous code mutates the set and we're relying on that fact here. That was super surprising to me...in fact, I was sure this line was a bug until I dug into the implementations. 

Even if it ends up being uglier, I'd very strongly prefer it if we didn't have hidden side-effects in the code like this.

Reply:
Yeah I mention it in a spec issue too :)

https://github.com/open-telemetry/opentelemetry-specification/pull/1318#discussion_r551854325

Do you see an alternative? Should I mutate a `handledExportNames` with `.add` instead?

Reply:
You could return an updated copy of the set, maybe?

Reply:
It's tricky since already returning a tracerprovider from one of them. It's a bit of duplication but I went with exposing the recognized names and checking eagerly instead of mutations.